### PR TITLE
refactor: email transactional middleware

### PR DESCRIPTION
**Pull request by tan-zx**
Miscellaneous refactoring while working on transactional SMS saving:
- Save one db roundtrip
- Remove unnecessary err handling
- Move db call up to make `acceptedAt` time more accurate
- Update schema in `openapi.yml` file: fix formatting + add explanation for enums
- Use `reply_to` instead of `replyTo` in returned JSON

### CODE REVIEW

**Comment by tan-zx** [comment]
Just curious re: the part excerpted below
- Currently, we have two separate (but identical schemas) for `application/json` and `multipart/form-data`
- Just curious, do we need to add another schema for `urlencoded`? Came across this while working on the body size limit PR

https://github.com/opengovsg/postmangovsg/blob/1ed3e978caedf612e850c5d5a44f3902d7ebab15/backend/openapi.yaml#L1670-1700

**Review by richard-yak — APPROVED** [review]
lgtm

File: backend/openapi.yaml [hunk 1]
```diff
@@ -4734,6 +4733,20 @@ components:
...
+
+            * `SENT` - the send request to our email provider was successful and our email provider will attempt to deliver the message to the recipient’s mail server
+
+            * `BOUNCED` - the recipient's mail server rejected the email
+
+            * `DELIVERED` - the email provider has successfully delivered the email to the recipient's mail server
+
+            * `OPENED` - the recipient received the message and opened it in their email client
+
+            * `COMPLAINT` - the email was successfully delivered to the recipient’s mail server, but the recipient marked it as spam
```
Comment (tan-zx) [inline backend/openapi.yaml:4749]: the relevant change is here!

Comment (tan-zx) [reply backend/openapi.yaml:4749]: so some weirdness I observed:
- on Swagger's [official website](https://swagger.io/docs/specification/data-models/enums/#:~:text=All%20values%20in%20an%20enum%20must%20adhere%20to%20the%20specified%20type.%20If%20you%20need%20to%20specify%20descriptions%20for%20enum%20items%2C%20you%20can%20do%20this%20in%20the%20description%20of%20the%20parameter%20or%20property%3A), they did not leave line breaks after each line of the enum description
- but if I didn't do that, the enum description is all clustered as a single sentence

image

image

**Pull request merged**